### PR TITLE
link 404 page to legacy site

### PR DESCRIPTION
**Pull request by rviscomi**
Features:
- higher res logo, with gradient "p"
- custom 404 error page with link to legacy subdomain
  - the link URL is the current page's URL with the host replaced by `legacy.httparchive.org`
- adjusted responsiveness of error page template

eg when visiting /trends.php on the new site, the user would see:

image

Progress on #33 and #13

### CODE REVIEW

**Comment by rviscomi** [comment]
Done!

Only big thing I changed is mapping `/interesting.php` to the State of the Web report's histogram view ([/reports/state-of-the-web?start=latest](https://beta.httparchive.org/reports/state-of-the-web?start=latest)). Looking at Google Analytics, these redirects should handle 2/3 of page views. We can also consider adding 302 redirects for unsupported pages like [/websites.php](https://legacy.httparchive.org/websites.php) or [/downloads.php](https://legacy.httparchive.org/downloads.php) to their legacy URL equivalents.

Also did the cookie thing with the banner:

image

File: legacy.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	def __init__(self, name, **kwargs):
+		self.name = name
+		self.kwargs = kwargs
+
+
+LEGACY_PATH_MAP = {
+	'/index.php': Page('index'),
+	'/about.php': Page('about'),
+	'/trends.php': Page('report', report_id='state-of-the-web'),
+	'/interesting.php': Page('report', report_id='state-of-the-web', start='latest')
```
Comment (igrigorik) [inline legacy.py]: Should we also point http://httparchive.org/downloads.php to some docs on GitHub, or some such? We can drop the MySQL bits, but it'd be nice to preserve guidance on how to access raw archives and BigQuery data, wdyt?

File: main.py [hunk 2]
```diff
@@ -160,7 +163,21 @@ def bad_request(e):
...
 @app.errorhandler(404)
 def page_not_found(e):
-	return render_template('error/404.html', error=e), 404
+	url = urlparse(request.url)
+	path = url.path
+	if legacy_util.should_redirect(path):
+		page = legacy_util.get_redirect_page(path)
+		redirect_url = url_for(page.name, **page.kwargs)
+		response = make_response(redirect(redirect_url, code=301))
+		# Set a cookie that expires 5 seconds after page load,
```
Comment (igrigorik) [inline main.py]: How confident are you that the page will load in under 5s? ;-)

File: templates/main.html [hunk 3]
```diff
@@ -10,6 +10,15 @@
 {% endblock %}
 
 {% block content %}
+	{% if 'legacy_welcome' in request.cookies %}
+		<div class="alert alert-warning alert-dismissable" role="alert">
+			<strong>Welcome to the new HTTP Archive!</strong>
```
Comment (igrigorik) [inline templates/main.html]: Link "new HTTP Archive" to the post as well?

File: legacy.py (hunk 1 above)
Comment (rviscomi) [reply legacy.py]: Should we redirect it to https://beta.httparchive.org/faq#how-do-i-use-bigquery-to-write-custom-queries-over-the-data ?

File: main.py (hunk 2 above)
Comment (rviscomi) [reply main.py]: Surprisingly no HTTP Archive data is available in CrUX for me to confirm, but I'm willing to accept the graceful degradation. :smile:

File: templates/main.html (hunk 3 above)
Comment (rviscomi) [reply templates/main.html]: Done

image

File: legacy.py (hunk 1 above)
Comment (igrigorik) [reply legacy.py]: Yeah, that looks good.

**Comment by rviscomi** [comment]
Added the redirect for /downloads.php. PTAL

**Pull request merged**